controller/src/api.py

```diff
@@ -67,33 +68,80 @@ async def get_auth_user(
         return AuthInfo(username="yhaviv@gmail.com", token=token)
 
 
+def send_to_application(path: str, method: str = "POST", request=None, auth=None, **kwargs):
+    """
+    Send a request to the application's API.
+
+    :param path:    The API path to send the request to.
+    :param method:  The HTTP method to use: GET, POST, PUT, DELETE, etc.
+    :param request: The FastAPI request object. If provided, the data will be taken from the body of the request.
+    :param auth:    The authentication information to use. If provided, the username will be added to the headers.
+    :param kwargs:  Additional keyword arguments to pass in the request function. For example, headers, params, etc.
+
+    :return:        The JSON response from the application.
+    """
+    url = f"{config.application_url}/api/{path}"
+
+    if isinstance(request, Request):
+        # If the request is a FastAPI request, get the data from the body
+        kwargs["data"] = request._body.decode("utf-8")
+    if auth is not None:
+        kwargs["headers"] = {"x_username": auth.username}
+
+    response = requests.request(
+        method=method,
+        url=url,
+        **kwargs,
+    )
+
+    # Check the response
+    if response.status_code == 200:
+        # If the request was successful, return the JSON response
+        return response.json()
+    else:
+        # If the request failed, raise an exception
+        response.raise_for_status()
+
+
 @router.post("/tables")
-async def create_tables(drop_old: bool = False, names: list[str] = None):
+def create_tables(drop_old: bool = False, names: list[str] = None):
     return client.create_tables(drop_old=drop_old, names=names)
 
 
 @router.post("/pipeline/{name}/run")
-async def run_pipeline(
-    request: Request, name: str, item: QueryItem, auth=Depends(get_auth_user)
+def run_pipeline(
+    request: Request, name: str, auth=Depends(get_auth_user)
 ):
     """This is the query command"""
-    app_server = request.app.extra.get("app_server")
-    if not app_server:
-        raise ValueError("app_server not found in app")
-    event = {
-        "username": auth.username,
-        "session_id": item.session_id,
-        "query": item.question,
-        "collection_name": item.collection,
+    return send_to_application(
+        path=f"pipeline/{name}/run",
+        method="POST",
+        request=request,
+        auth=auth,
+    )
+
+
+@router.post("/collections/{collection}/{loader}/ingest")
+def ingest(
```

Comment:
ingestion pipelines are yet another pipeline, so I'm not sure why we should support `ingest` and `run_pipeline` and not just the run pipeline. We need to think about it.

Reply:
The ingest pipeline is working locally by a data loader that we implemented or a `langchain` document loader, not done by `storey` steps.

Reply:
well.. this is temporary for now.